### ISSUE TITLE
500 Internal Server Error: Unable to find Form Request 

### ISSUE THREAD

**Issue by el-schneider** [damcclean/simple-commerce#740]
### What's the issue you're running into?

Hi, I'm running into an error when trying to update the cart through Ajax. However the cart actually does get updated.

### Steps to reproduce

The `_request: "Empty",` I don't really understand but I took it from the provided template tags.

My Javascript:

```js
await axios.post("/!/simple-commerce/cart-items", {
    _token: this.token,
    product: id,
    quantity,
    variant,
    _request: "Empty",
});
```

Payload looks like this:
```json
{
    "_token": "qZrinAckUTVentwYa5kex6lvv43zJa9l9O4PGkUp",
    "product": "3b751903-373a-486d-bcab-49ed857b2e4b",
    "quantity": 1,
    "variant": "sm",
    "_request": "Empty"
}
```

Response like this:
```json
{
    "message": "Unable to find Form Request [http://myproject.test/shop]",
    "exception": "Exception",
    "file": "/Users/jonas/dev/nandms/myproject/vendor/doublethreedigital/simple-commerce/src/Http/Requests/AcceptsFormRequests.php",
    "line": 37,
    "trace": [
        {
            "file": "/Users/jonas/dev/nandms/myproject/vendor/doublethreedigital/simple-commerce/src/Http/Requests/CartItem/StoreRequest.php",
            "line": 42,
            "function": "buildFormRequest",
            "class": "DoubleThreeDigital\\SimpleCommerce\\Http\\Requests\\CartItem\\StoreRequest",
            "type": "->"
        },
        {
            "file": "/Users/jonas/dev/nandms/myproject/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php",
            "line": 36,
            "function": "rules",
            "class": "DoubleThreeDigital\\SimpleCommerce\\Http\\Requests\\CartItem\\StoreRequest",
            "type": "->"
        },
        {
            "file": "/Users/jonas/dev/nandms/myproject/vendor/laravel/framework/src/Illuminate/Container/Util.php",
            "line": 41,
            "function": "Illuminate\\Container\\{closure}",
            "class": "Illuminate\\Container\\BoundMethod",
            "type": "::"
        },
…
    ]
}
```

### Environment

Application Name: MYPROJECT
Laravel Version: 9.34.0
PHP Version: 8.1.11
Composer Version: 2.4.2
Environment: local
Debug Mode: ENABLED
URL: myproject.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Simple Commerce
Gateways: Stripe, PayPal
Repository: Customer: DoubleThreeDigital\SimpleCommerce\Customers\EntryCustomerRepository
Repository: Order: DoubleThreeDigital\SimpleCommerce\Orders\EntryOrderRepository
Repository: Product: DoubleThreeDigital\SimpleCommerce\Products\EntryProductRepository
Shipping Methods: Standard Post
Tax Engine: DoubleThreeDigital\SimpleCommerce\Tax\BasicTaxEngine

Statamic
Addons: 4
Antlers: runtime
Version: 3.3.43 PRO

Statamic Addons
aryehraber/statamic-impersonator: 2.4.1
doublethreedigital/simple-commerce: 4.1.0
octoper/statamic-html-minify: 1.0.0
spatie/statamic-responsive-images: 2.14.0

**Comment by duncanmcclean** [comment]
What happens if you don't provide a `_request` parameter?

**Comment by el-schneider** [comment]
The same thing

EDIT: actually no. I doesn't get updated anymore but still fails with the same error.

**Comment by el-schneider** [comment]
It might be SSL-related. It seems to work fine on my staging-server but not locally.

**Comment by duncanmcclean** [comment]
I've just taken a look at this...

Even when I make a request where the `_request` parameter and the value is `Empty`, it saves & responds as expected. If you don't provide it, it'll just be set to `Empty` behind the scenes anyway.

In the file where the exception is coming from, it essentially checks what you've provided in the `_request` parameter to see if it can find a Laravel Form Request with that name (form requests are used for validation).

It looks like, for some reason, you _might_ be telling Simple Commerce to look for a form request called `http://myproject.test/shop` which won't exist. Although, I'm not 100% where since it doesn't look that way from your code.

I don't suppose you'd be able to send me a link to the staging site / the repository and I can have a play around to see if I can figure out what might be causing the issue?

**Comment by duncanmcclean** [comment]
Also, do you have this set in your config anywhere? `disable_form_parameter_validation`

**Comment by el-schneider** [comment]
I sent you an invite. `disable_form_parameter_validation` is set, yes.

**Comment by duncanmcclean** [comment]
Thanks! I’ll take a look when I can.

**Comment by el-schneider** [comment]
Appreciate it! It might be related to custom routes that I have set up for `/shop` and `/cart`

**Comment by duncanmcclean** [comment]
Hey! I've just pulled down your repository locally (using the `dev` branch) and I'm still not able to reproduce.

When I add products to the cart on the `/shop` page, the products are added absolutely fine. No error - it works as I'd expect it to..

I've added SSL to my local site which didn't cause any issues. 

Have you tried clearing all of the caches to check if something cached could be what's causing the issues for you locally?

`php artisan cache:clear`
`php artisan config:clear`
`php artisan route:clear`
